src/web/lib/contributions.tsx

```diff
@@ -0,0 +1,92 @@
+    'gu.contributions.contrib-timestamp';
+
+// Cookie set by the User Attributes API upon signing in.
+// Value computed server-side and looks at all of the user's active products,
+// including but not limited to recurring & one-off contributions,
+// paper & digital subscriptions, as well as user tiers (GU supporters/staff/partners/patrons).
+// https://github.com/guardian/members-data-api/blob/3a72dc00b9389968d91e5930686aaf34d8040c52/membership-attribute-service/app/models/Attributes.scala
+export const shouldShowSupportMessaging = (): boolean => {
+    const hideSupportMessaging =
+        getCookie(HIDE_SUPPORT_MESSAGING_COOKIE) === 'true';
+
+    return !hideSupportMessaging;
+};
+
+// Determine if user is a recurring contributor by checking if they are signed in
+// AND have at least one of the relevant cookies.
+// We need to look at both User Attributes and Frontend Support cookies
+// as the former might not reflect the latest contributor status, since it's set upon signing in.
+// Frontend Support cookies are set when a contribution is made.
+export const isRecurringContributor = (isSignedIn: boolean): boolean => {
+    // Attributes cookie - we want this to have a specific value
+    const isRecurringContributorFromAttrs =
+        getCookie(RECURRING_CONTRIBUTOR_COOKIE) === 'true';
+
+    // Support cookies - we only care whether these exist
+    const hasMonthlyContributionCookie =
+        getCookie(SUPPORT_RECURRING_CONTRIBUTOR_MONTHLY_COOKIE) !== null;
+    const hasAnnualContributionCookie =
+        getCookie(SUPPORT_RECURRING_CONTRIBUTOR_ANNUAL_COOKIE) !== null;
+
+    return (
+        isSignedIn &&
+        (isRecurringContributorFromAttrs ||
+            hasMonthlyContributionCookie ||
+            hasAnnualContributionCookie)
+    );
+};
+
+// looks at attribute and support cookies
+// ONE_OFF_CONTRIBUTION_DATE_COOKIE (attributes cookie, when loggin in)
+// SUPPORT_ONE_OFF_CONTRIBUTION_COOKIE (support cookie, when making one-off contribution)
+// Get the date of the latest one-off contribution by looking at the two relevant cookies
+// and returning a Unix epoch string of the latest date found.
+export const getLastOneOffContributionDate = (): string => {
+    // Attributes cookie - expects YYYY-MM-DD
+    const contributionDateFromAttributes = getCookie(
+        ONE_OFF_CONTRIBUTION_DATE_COOKIE,
+    );
+
+    // Support cookies - expects Unix epoch
+    const contributionDateFromSupport = getCookie(
+        SUPPORT_ONE_OFF_CONTRIBUTION_COOKIE,
+    );
+
+    try {
+        // Parse dates into common format so they can be compared
+        const parsedDateFromAttributes = contributionDateFromAttributes
+            ? Date.parse(contributionDateFromAttributes)
+            : 0;
+        const parsedDateFromSupport = contributionDateFromSupport
+            ? parseInt(contributionDateFromSupport, 10)
+            : 0;
+
+        // Determine the latest of two dates
+        const latestOnOffContributionDate = Math.max(
```

Comment:
Not necessarily better, but what about something like:

```if (parsedDateFromAttributes > parsedDateFromSupport) {
    return parsedDateFromAttributes
} else {
    return parsedDateFromSupport
}```

Reply:
And add a guard beforehand by the cookies to exit early if both are null? I.e.

```
if (!contributionDateFromAttributes && !contributionDateFromSupport) {
    return undefined;
}
```

Reply:
Yeah I think your suggestion is better. I've refactored to use your approach, with some extra checks to make sure we never return `NaN` if that's what's returned by the parsing of either date. I think this lets us get rid of the try catch as well (unless I'm missing  something), I can't see a scenario where an error would be thrown with the existing logic @nic-guardian can you think of one? 🤔